internal/resources/fetcher.go

```diff
@@ -73,7 +71,7 @@ func (f *Fetcher) GetResourcesForPolicies(ctx context.Context, policies []polici
 	for resourceFilter, policies := range gvrMap {
 		isNamespaced, err := f.isNamespacedResource(resourceFilter.groupVersionResource)
 		if err != nil {
-			if errors.Is(err, constants.ErrResourceNotFound) {
+			if apimachineryerrors.IsNotFound(err) {
```

Comment:
From what I can see, this is the actual fix. The API server wasn't returning the error type we were expecting.

Has that always been the case, or has the API server changed the error type it returns?
Should we leave the old error check around?

Reply:
I do not remember why we were not using the apimachinery before. But I have the impression that new code is doing the same but using the k8s lib.

Reply:
The `isNamespacedResource()` function doesn't return any API server errors, as what it does is to obtain all the known GVKs, and then iterate ours through them to see if ours are namespaced. This may fail if we have bogus or CRD resources defined in policy rules.

The fix is a change in `isNamespacedResource()` function to return apimachinery's `NotFound` error instead of our `ErrResourceNotFound`, and then I only do 1 `if apimachineryerrors.IsNotFound(err)` instead of checking once for the apimachinery error and then again for our own error.

Our `ErrResourceNotFound` is returned in other places, particularly in the store functions.

Reply:
Now I undertstand what is going on. I'm totally fine with this approach, marking the comment as solved.